docs/users/installation.md

```diff
@@ -0,0 +1,225 @@
+          </compilerPlugins>
+          <addScalacArgs>-Yrangepos|-Ywarn-unused-import|-P:semanticdb:sourceroot:/path/to/root-directory</addScalacArgs>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+</profile>
+```
+
+Make sure to replace `/path/to/root-directory` with the absolute path to the
+root directory of Maven build. This directory must match the directory from
+where you invoke the `scalafix` command-line interface.
+
+Next, compile your project with the `scalafix` profile
+
+```
+mvn clean test -DskipTests=true -P scalafix
+```
+
+We compile both main sources and tests to have semantic information generated
+for all of them, but we skip test execution because it is not the point of that
+compilation. The added flags for scala are given with the `addScalaArgs` option
+(see http://davidb.github.io/scala-maven-plugin/help-mojo.html#addScalacArgs):
+
+- `-Yrangepos` is required for `semanticdb` to function properly,
+- `-Xplugin:PLUGIN/semanticdb-scalac_@SCALA212@-@SCALAMETA@.jar` give the path
+  where the `semanticdb` jar can be found,
+- (optional) `-Xplugin-require:semanticdb` tells scalac to fails if it can't
+  load the `semanticdb` plugin,
+- (optional) `-Ywarn-unused-import` is required for the `RemoveUnusedImports`
+  rule. If you don't run `RemoveUnusedImports` you can skip this flag. Consult
+  the scalafix documentation for each rule to see which flags it requires.
+
+After compilation, double check that there exists a directory
+`target/classes/META-INF/semanticdb/` containing files with the `.semanticdb`
+extension.
+
+Note that will need to recompile to get up-to-date `semanticdb` information
+after each modification.
+
+Next, [install the command line](#command-line). Finally, invoke the `scalafix`
+command line interface from the same directory as the value of
+`-P:semanticdb:sourceroot:/path/to/root-directory`
+
+```sh
+scalafix --rules RemoveUnusedImports
+```
+
+**NOTE**. This setup will not work for some semantic rules like
+`ExplicitResultTypes` that require a complete `--classpath` flag with the full
+classpath of your application. Help with creating a Scalafix Maven plugin or
+updating this guide for how to pass `--classpath` would be appreciated!
+
+## Pants
+
+Scalafix support is built into Pants and will run scalafmt after running
+scalafix rewrite rules to maintain your target formatting. Usage instructions
+can be found at https://www.pantsbuild.org/scala.html
+
+## SNAPSHOT
+
+Our CI publishes a snapshot release to Sonatype on every merge into master. Each
+snapshot release has a unique version number, jars don't get overwritten. To
+find the latest snapshot version number, go to
+<https://oss.sonatype.org/content/repositories/snapshots/ch/epfl/scala/scalafix-core_2.12/>
+and select the version number at the bottom, the one with the latest "Last
+Modified". Once you have found the version number, adapting the version number
+
+If using the sbt plugin
+
+```scala
+// project/plugins.sbt
```

Comment:
We can get this version with a small javascript snippet: https://github.com/scalacenter/bloop/pull/374

Reply:
Cool, we can do that in a separate PR